ipi/engine/forcefields.py

```diff
@@ -375,6 +375,96 @@ def evaluate(self, r):
         r["status"] = "Done"
 
 
+try:
+    import quippy
+except Exception as e:
+    quippy = None
+    quippy_exc = e
+
+
+class FFQUIP(ForceField):
+
+    """Basic fully pythonic force provider.
+
+    Computes an arbitrary interaction potential implemented in QUIP. 
+    Parallel evaluation with threads.
+
+    Attributes:
+        parameters: A dictionary of the parameters used by QUIP. Of the
+            form {'name': value}.
+        requests: During the force calculation step this holds a dictionary
+            containing the relevant data for determining the progress of the step.
+            Of the form {'atoms': atoms, 'cell': cell, 'pars': parameters,
+                         'status': status, 'result': result, 'id': bead id,
+                         'start': starting time}.
+    """
+
+    def __init__(self, latency=1.0e-3, name="", pars=None, dopbc=True):
+        """Initialises QUIP.
+
+        Args:
+        pars: Mandatory dictionaru, giving the parameters needed by QUIP.
+        """
+        if quippy is None:
+            raise ImportError("QUIPPY import failed due to expection : " + str(e))
+
+        # a socket to the communication library is created or linked
+        super(FFQUIP, self).__init__(latency, name, pars, dopbc)
+        self.init_file = str(self.pars["init_file"])
+        self.args_str = str(self.pars["args_str"])
+        self.param_file = str(self.pars["param_file"])
+
+        # Initializes an atoms object and the interaction potential
+        self.atoms = quippy.Atoms(self.init_file)
+        self.pot = quippy.Potential(self.args_str, param_filename=self.param_file)
+       
+        # Initializes the conversion factors from i-pi to QUIP
+        self.len_conv = 0.529177 
+        self.energy_conv = 27.211386
+
+    def poll(self):
```

Comment:
Make sure this is actually needed. Also after this is merged please take care of merging master into feat/fast_ff because I suspect this will interact with the new default strategy which is not to thread over direct calls